src/iop/channelmixerrgb.c

```diff
@@ -3854,11 +3854,16 @@ void reload_defaults(dt_iop_module_t *module)
       g->delta_E_label_text = NULL;
     }
 
+    const int pos = dt_bauhaus_combobox_get_from_value(g->illuminant, DT_ILLUMINANT_CAMERA);
     if(dt_image_is_matrix_correction_supported(img) && !dt_image_is_monochrome(img))
-      // add back if needed
-      dt_bauhaus_combobox_set_from_value(g->illuminant, DT_ILLUMINANT_CAMERA);
+    {
+      if(pos == -1)
```

Comment:
you can combine this conditional to the parent `if`.

Reply:
I have to admit I almost did, but that would mean that the "else" would be executed whenever the DT_ILLUMINANT_CAMERA is there, removing it even if it should stay. Or am I misunderstanding you?

Reply:
Ah true! My mistake.